pkg/cmd/server/kubernetes/node_config.go

```diff
@@ -214,6 +216,18 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig, enableProxy, enable
 		return nil, err
 	}
 
+	// Initialize cloud provider
+	if len(server.CloudProvider) > 0 && server.CloudProvider != v1alpha1.AutoDetectCloudProvider {
```

Comment:
I added some simple unit tests for server.CloudProvider parsing, however there is not a single unit test that would test whole `BuildKubernetesNodeConfig`. This function does many decisions that are not tested anywhere and building a cloud provider is just one of its duties.

IMO, there should be e2e tests that would catch GCE or AWS volume plugin throwing errors when mounting a volume.